packages/ui/src/components/IconButton/IconButton.tsx

```diff
@@ -18,7 +18,8 @@ export const iconButtonStyles = cva([], {
   },
 });
 
-const iconSize: Record<ButtonSizeProp, number> = {
+type Size = Required<Pick<ButtonProps, "size">>;
```

Comment:
Because for this mapping, we're placing the `size` as mandatory (because there's no mapping to `undefined` in this map) and it's optional from the `ButtonProps`.

So we need to tell Typescript this object map won't receive an `undefined` value.